nodebuilder/header/opts.go

```diff
@@ -0,0 +1,20 @@
+package header
+
+import (
+	header "github.com/celestiaorg/celestia-node/header"
+	libhead "github.com/celestiaorg/celestia-node/libs/header"
+	p2p "github.com/celestiaorg/celestia-node/libs/header/p2p"
+)
+
+// WithMetrics provides sets `MetricsEnabled` to true on ClientParameters for the header exchange
+func WithMetrics(ex libhead.Exchange[*header.ExtendedHeader]) error {
+	exchange, ok := (ex).(*p2p.Exchange[*header.ExtendedHeader])
+	if !ok {
+		// not all implementations of libhead.Exchange[*header.ExtendedHeader]
```

Comment:
It's fine to panic in this case because when WithMetrics is provided, we always expect it to be the type we want. Yeah, there may be other implementations, but if it's given here, it means something went wrong with the program, so panicking earlier is helpful.

Reply:
The thing is, we received multiple implementations of `libhead.Exchange` interface, right, so we want to only impact a specific `implementer. I mean just from reading what I just wrote, I really dislike how I am doing it, but the other solution to avoid this, would be to add `RegisterMetrics() error` to the interface and have all implementers define it. That's somewhat impactful in terms of code.

This quick hack will absolve us from that. So in summary, we got multiple implementers, and we only want a specific implementer that defines `RegisterMetrics() error` thus this ugliness.

Reply:
I understand, but that does not explain why we shouldn't panic there.